### PR TITLE
[stable/owncloud] Set deployment strategy with values

### PR DESCRIPTION
**Pull request by zabudskyi**
Signed-off-by: Pavlo Zabudskyi <pavlo.zabudskyi@gmail.com>

#### What this PR does / why we need it:
We need possibility to set `.spec.strategy.type` to `Recreate` for owncloud deployment. When PVC access mode is set to `ReadWriteOnce` and pod is deleted, it can't re-map volume. Default value is set to `RollingUpdate`.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/owncloud/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: owncloud
-version: 8.0.1
+version: 8.0.2
```
Comment (alemorcuq) [inline stable/owncloud/Chart.yaml]: The version should be a new minor.

File: stable/owncloud/values.yaml [hunk 2]
```diff
@@ -218,6 +218,9 @@ persistence:
     accessMode: ReadWriteOnce
     size: 8Gi
 
+# Update strategy type (e.g. Recreate or RollingUpdate)
+updateStrategy: RollingUpdate
+
```
Comment (alemorcuq) [inline stable/owncloud/values.yaml]: I would do this instead, allowing for more customization:

```yaml
## Set up update strategy for the ownCloud installation.
## ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
## Example:
# updateStrategy:
#  type: RollingUpdate
#  rollingUpdate:
#    maxSurge: 25%
#    maxUnavailable: 25%
updateStrategy:
  type: RollingUpdate
```

File: stable/owncloud/README.md [hunk 3]
```diff
@@ -95,6 +95,7 @@ The following table lists the configurable parameters of the ownCloud chart and
 | `persistence.owncloud.existingClaim`| An Existing PVC name for ownCloud volume   | `nil` (uses alpha storage class annotation)             |
 | `persistence.owncloud.accessMode`   | PVC Access Mode for ownCloud volume        | `ReadWriteOnce`                                         |
 | `persistence.owncloud.size`         | PVC Storage Request for ownCloud volume    | `8Gi`                                                   |
+| `updateStrategy`                    | Owncloud deployment strategy               | `RollingUpdate`                                         |
```
Comment (alemorcuq) [inline stable/owncloud/README.md]: I would change the default value to:

```Markdown
`{type: "RollingUpdate"}`
```

See the other comments.

File: stable/owncloud/templates/deployment.yaml [hunk 4]
```diff
@@ -14,6 +14,8 @@ spec:
       app: {{ template "owncloud.fullname" . }}
       release: "{{ .Release.Name }}"
   replicas: 1
+  strategy:
+    type: {{ .Values.updateStrategy }}
```
Comment (alemorcuq) [inline stable/owncloud/templates/deployment.yaml]: I would do this instead, allowing for more customization:

```yaml
{{- if .Values.updateStrategy }}
strategy: {{ toYaml .Values.updateStrategy | nindent 4 }}
{{- end }}
```

**Review by alemorcuq — CHANGES_REQUESTED** [review]

**Comment by alemorcuq** [comment]
Thank you for your contribution, @zabudskyi. I've left some comments on what I think would be a better implementation.

**Comment by zabudskyi** [comment]
@alemorcuq 
Thank you for your feedback. I updated PR. I added Signed-off part for merge commit as well.

**Comment by alemorcuq** [comment]
Thanks, @zabudskyi! The DCO is missing, see this link for how to fix it: https://github.com/helm/charts/pull/20083/checks?check_run_id=391038032

**Comment by zabudskyi** [comment]
I'll open new PR

**Pull request closed without merging**